### DISCUSSION TITLE
How do I load an existing *.gpx track file into OsmAnd let it show on map?

### DISCUSSION THREAD

**Discussion by pstein** [osmandapp/OsmAnd#13931] [Q&A]
How do I load an existing *.gpx track file into OsmAnd let it show on map?

**Comment by pebogufi** [comment]
Open the file with a file explorer (just tap on the file) then on "open with" select OsmAnd.

**Reply by Lee-Carre** [reply]
If I recall (risky, I know(!)), there's a specific directory which OsmAnd uses for tracks.

Aside from importing via an intent, one could also copy the desired GPX file to that directory(, and probably have to relaunch OsmAnd; I've not tried such things, before).

**Reply by tommycrock** [reply]
> one could also copy the desired GPX file to that 
directory

I don't think this will work for most cases on Android 11. Even if a user can access their osmand directory, osmand won't be able to read the files placed there (I don't know if there are exceptions)

**Reply by Lee-Carre** [reply]
> on Android 11

Ah, yes. Quite right. I forgot about this new devilry. I'm still on v10.
Similar to #13541, then.

Though, if #13538 were implemented, that might avoid that whole mess.

**Reply by mariush444** [reply]
if your net.osmand.plus directory is in Android/media (instead data) then it is still possible even on Android 11

**Reply by tommycrock** [reply]
Interesting. It doesn't work in download for me: I can read and write the files, but osmand can't read them if I write them.

**Reply by mariush444** [reply]
@tommycrock  is to to me?
I didn't write about `download` but about .`../Android/media/net.osmand.plus/files/`

**Reply by tommycrock** [reply]
@mariush444 yes, I'm saying the difference between them is interesting and worth knowing. I'm not planning to try swapping to media in case it doesn't work for me - it was such a pain importing everything last time - but if things go wrong or I need to reinstall I'll be trying it.

**Reply by mariush444** [reply]
@tommycrock yes, the moving/swapping is a real pain but any way this path is working OK.
I have confirmation from 3 other users.

**Comment by pstein** [comment]
Thank you so far for the comments but I am confused about the handling in Android 11 and 12:

If I understand correctly users can copy *.gpx files (via USB cable) to the directory Android/media/net.osmand.plus/files/ and Osmand will be able to read them (WITHOUT additional permissions "All files access").

If I plan to save the *.gpx files in another directory I can do it but give "All files access" permission to OsmAnd.

BUT: here I have no such directory as

Android/media/net.osmand.plus/files/

The only related folder I found so far is

Android/data/Installed Apps/net.osmand/

but OsmAnd seems not to be able to read (automatically) *.gpx files from there.

Whats wrong?

**Reply by mariush444** [reply]
You can create directory Android/media/net.osmand.plus in main memmory or in external SD card using file manager.
I use Total Commander, other users use MIXplorer or anotherer tool probably.
If you don't like external tool you can just set directory path in osmand
Settings -> osmand settings -> Data storage folder -> manualy specified 

BUT
I do know nothing about Android 12 and I do not care about  "All files access"
